src/NosCore.MasterServer/Controllers/MailController.cs

```diff
@@ -192,26 +194,22 @@ public async Task<bool> SendMail([FromBody] MailRequest mail)
 
                 mail.Amount = (it.Type == NoscorePocketType.Etc) || (it.Type == NoscorePocketType.Main) ? mail.Amount
                     : 1;
-                itemInstance = _itemProvider.Create((short) mail.VNum, mail.Amount ?? 1, mail.Rare ?? 0,
+                itemInstance = _itemProvider.Create((short)mail.VNum, mail.Amount ?? 1, mail.Rare ?? 0,
                     mail.Upgrade ?? 0);
-                if (itemInstance == null)
-                {
-                    return false;
-                }
 
                 _itemInstanceDao.InsertOrUpdate(ref itemInstance);
                 mailref.ItemInstanceId = itemInstance.Id;
             }
 
-            var receiver = await _connectedAccountHttpClient.GetCharacter(mailref.ReceiverId, null).ConfigureAwait(false);
-            var sender = await _connectedAccountHttpClient.GetCharacter(mailref.SenderId, null).ConfigureAwait(false);
+            var receiver = await _connectedAccountHttpClient.GetCharacterAsync(mailref.ReceiverId, null).ConfigureAwait(false);
+            var sender = await _connectedAccountHttpClient.GetCharacterAsync(mailref.SenderId, null).ConfigureAwait(false);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused local variable 'sender'.](https://app.codacy.com/manual/NosCoreIO/NosCore/pullRequest?prid=5266673)